tests/drivers/spi/spi_loopback/src/spi.c

```diff
@@ -582,51 +597,6 @@ ZTEST(spi_loopback, test_spi_loopback)
 		goto end;
 	}
 
-#if DT_HAS_COMPAT_STATUS_OKAY(st_stm32h7_spi)
```

Comment:
To reduce complexity in this file, instead of having another set of tests for 16 bits frames, use overlay and conf files (as DMA does)